feat(studio): simplify query performance filters

kemaldotearth opened the pull request:
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

This PR is simplifying our Query Performance tabs filter bar. We have a redundant sort dropdown as well as a weird way to find queries in the table. Adding some simplicity to it and removing some noise.

kemaldotearth committed:
feat: remove redundant sort by

We had conflicting sort by dropdown which clashed with the individual column sorting itself. Either we do something similar to our table editor or leave it out for now. Opting to leave it out for now.

kemaldotearth committed:
fix: alignment of download dropdown

This aligns the download dropdown on the Query Performance page to look visually correct.

charislam reviewed (approved):
LGTM

kemaldotearth committed:
feat: improved query search

This improves our query search in the Query Performance area. Now it does a local search through the available queries in the table. I removed it requesting the string from server for now. Simple

:x

q

kemaldotearth committed:
fix: increased width of filter popover

The roles filter popover is was too narrow for its contents. Increased teh size slightly.

kemaldotearth committed:
chore: remove commented out component

kemaldotearth committed:
chore: update placeholder for query filter

kemaldotearth committed:
fix: leftover prop on query performance page

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-simplify-filters

charislam reviewed (commented):
Question about the change to local search: I really like the new behavior, but I'm wondering if anyone actually depends on the network search? E.g., I have a very specific query (or set of queries) I want to track performance for, they don't actually show up in the top 20, but in the old behavior, I can query for them from the backend. In the new behavior, I can't fetch them at all.

kemaldotearth commented on apps/studio/components/interfaces/QueryPerformance/hooks/useIndexInvalidation.ts:
```diff
@@ -34,7 +34,6 @@ export function useIndexInvalidation() {
   const roles = router?.query?.roles ?? []
 
   const queryPerformanceQuery = useQueryPerformanceQuery({
-    searchQuery,
```
Good point re: search, we can revert it back to finding via query. To me it felt a little slow and unreliable to find things inline, but maybe not such an issue? Let me give it a try (and yes, we can remove `searchQuery`!)

kemaldotearth replied:
Try again @charislam - bonus you can now also copy over the url (like before) and share the filtered search too.

kemaldotearth committed:
feat: reintroduce sql based query for search

Bringing back a hybrid approach - seems to work fine but also returns items that are not in your visible queries.

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-simplify-filters

joshenlim commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceFilterBar.tsx:
```diff
@@ -59,73 +45,39 @@ export const QueryPerformanceFilterBar = ({
+            autoComplete="off"
+            icon={<Search size={12} />}
+            value={searchQuery}
+            onChange={(e: any) => onSearchQueryChange(e.target.value)}
```
also avoid usage of `any` as much as possible

joshenlim commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceGrid.tsx:
```diff
@@ -168,7 +168,21 @@ export const QueryPerformanceGrid = ({ queryPerformanceQuery }: QueryPerformance
+    }
+
+    const searchTerm = search.toLowerCase()
+    return allData.filter((row: any) => {
+      // Search in the query text
+      const queryText = row.query?.toLowerCase() || ''
+      return queryText.includes(searchTerm)
```
hmm is there any context as to why we're moving away from server side searching to client side?

joshenlim replied:
JFYI as well - with this we're currently actually doing _both_ server and client side searching ([ref](https://github.com/supabase/supabase/blob/feat/query-perf-simplify-filters/apps/studio/pages/project/%5Bref%5D/advisors/query-performance.tsx#L46)) as the search query is also being passed into the RQ hook that retrieves the report data

joshenlim replied:
side thought - im thinking this could get confusing from a UX POV (although i might be over worrying too)

sorting is done on the server side, but searching is done on the client side (based on the current data that's shown) - could this be confusing? wdyt?

kemaldotearth replied:
I think we're back to that with the latest changes. Server side search, last 20 results.

joshenlim commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceFilterBar.tsx:
```diff
@@ -59,73 +45,39 @@ export const QueryPerformanceFilterBar = ({
-    }
-  }
-
-  const onSortChange = (order: 'asc' | 'desc') => {
```
Unrelated, but I actually didn't realise that the columns were sortable by clicking on the header - wondering if we should add a tooltip or some UI indication

kemaldotearth replied:
I'll do a follow up updating the UI of the column headers. I think we should use the same pattern as Tables editor cols etc. I used this also in the Query Insights.

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-simplify-filters

kemaldotearth committed:
feat: use debounce on search query

kemaldotearth committed:
feat: use nuqs instead of next router for roles

kemaldotearth commented on apps/studio/components/interfaces/QueryPerformance/QueryPerformanceFilterBar.tsx:
```diff
@@ -35,21 +27,15 @@ export const QueryPerformanceFilterBar = ({
+  })
 
-  const [searchInputVal, setSearchInputVal] = useState(defaultSearchQueryValue)
+  const defaultFilterRoles = router.query.roles ? (router.query.roles as string[]) : []
```
next/router is yesterdays news, seeee ya.

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-simplify-filters

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-simplify-filters

joshenlim committed:
Merge branch 'master' into feat/query-perf-simplify-filters

joshenlim committed:
Tiny optimization for debounced search to immediately run query if clearing input

joshenlim committed:
Add clear search CTA

joshenlim committed:
Remove client side searching

joshenlim committed:
Revert changes in useIndexInvalidation

joshenlim reviewed (approved):
Verified changes locally

kemaldotearth committed:
Merge remote-tracking branch 'origin/master' into feat/query-perf-simplify-filters
